src/maven/background_peaks_update.h

```diff
@@ -113,6 +113,12 @@ class BackgroundPeakUpdate : public QThread
 
    Classifier* clsf;
 
+   //peaks search parameters
+   shared_ptr<PeaksSearchParameters> peaksSearchParameters = shared_ptr<PeaksSearchParameters>(new PeaksSearchParameters());
+
+   //Issue 606: mzkitchen-specific parameters
+   shared_ptr<LCLipidSearchParameters> lipidSearchParameters = shared_ptr<LCLipidSearchParameters>(new LCLipidSearchParameters());
+   shared_ptr<MzkitchenMetaboliteSearchParameters> mzkitchenMetaboliteSearchParameters = shared_ptr<MzkitchenMetaboliteSearchParameters>(new MzkitchenMetaboliteSearchParameters());
 
```

Comment:
I found that saving the parameters as fields in the `BackgroundPeaksUpdate` class was the easiest way. the `PeakDetectionDialog` generates a `BackgroundPeaksUpdate` class instance, which performs the update.  It is called `Background` because it executes in a background thread (off of the main GUI thread).